packages/cli/src/commands/store.js

```diff
@@ -1,18 +1,92 @@
 /* global process */
 import fs from 'fs';
 import os from 'os';
+import Buffer from 'buffer';
 
 import { makeNodeReader } from '@endo/stream-node';
 import { makeReaderRef } from '@endo/daemon';
 import { E } from '@endo/far';
 
 import { withEndoAgent } from '../context.js';
 
-export const store = async ({ storablePath, name, agentNames }) => {
-  const nodeReadStream = fs.createReadStream(storablePath);
-  const reader = makeNodeReader(nodeReadStream);
-  const readerRef = makeReaderRef(reader);
-  await withEndoAgent(agentNames, { os, process }, async ({ agent }) => {
-    await E(agent).store(readerRef, name);
-  });
+export const store = async ({
+  name,
+  agentNames,
+  storePath,
+  storeStdin,
+  storeText,
+  storeTextStdin,
+  storeJson,
+  storeJsonStdin,
+  storeBigInt,
+}) => {
+  const modes = {
+    storePath,
+    storeStdin,
+    storeText,
+    storeTextStdin,
+    storeJson,
+    storeJsonStdin,
+    storeBigInt,
+  };
+  const selectedModes = Object.entries(modes).filter(
+    ([_modeName, value]) => value !== undefined,
+  );
+  const selectedModeNames = selectedModes.map(([modeName]) => modeName);
+  if (selectedModes.length !== 1) {
+    // Usage error should be reported without trace.
+    // eslint-disable-next-line no-throw-literal
+    throw `Must provide exactly one store flag. Got flags for: (${selectedModeNames.join(
+      ', ',
+    )})`;
+  }
+
+  if (storePath !== undefined) {
+    const nodeReadStream = fs.createReadStream(storePath);
+    const reader = makeNodeReader(nodeReadStream);
+    const readerRef = makeReaderRef(reader);
+    await withEndoAgent(agentNames, { os, process }, async ({ agent }) => {
+      await E(agent).storeBlob(readerRef, name);
+    });
+  } else if (storeStdin) {
+    const reader = makeNodeReader(process.stdin);
+    const readerRef = makeReaderRef(reader);
+    await withEndoAgent(agentNames, { os, process }, async ({ agent }) => {
+      await E(agent).storeBlob(readerRef, name);
+    });
+  } else if (storeText !== undefined) {
+    await withEndoAgent(agentNames, { os, process }, async ({ agent }) => {
+      await E(agent).storeValue(storeText, name);
+    });
+  } else if (storeTextStdin) {
+    const reader = makeNodeReader(process.stdin);
+    const chunks = [];
+    for await (const chunk of reader) {
+      chunks.push(chunk);
+    }
+    const bytes = Buffer.concat(chunks);
+    const text = new TextDecoder().decode(bytes);
+    await withEndoAgent(agentNames, { os, process }, async ({ agent }) => {
+      await E(agent).storeValue(text, name);
+    });
+  } else if (storeJson !== undefined) {
```

Comment:
Some of these mode conditions are expressed as `if (modeVariable)` while others are `if (modeVariable !== undefined)`. We should stick to one or the other.